plugins/modules/zos_backup_restore.py

```diff
@@ -1005,12 +991,16 @@ def to_dunzip_args(**kwargs):
             size += kwargs.get("space_type")
         zoau_args["size"] = size
 
-    if kwargs.get("hlq"):
-        zoau_args["high_level_qualifier"] = kwargs.get("hlq")
-
     if kwargs.get("tmp_hlq"):
         zoau_args["tmphlq"] = str(kwargs.get("tmp_hlq"))
 
+    if kwargs.get("hlq") is None:
+        zoau_args["high_level_qualifier"] = datasets.get_hlq()
+    elif kwargs.get("hlq") == "K" or kwargs.get("hlq") == "R":
+        zoau_args["keep_original_hlq"] = True
+    elif kwargs.get("tmp_hlq"):
```

Comment:
i think this `elif` might be a nop (or worse, it might create unexpected hlqs). From the way im understanding this if-elif-elif:
- (1) if "hlq" is not defined by the user, 
  - default behavior - which is running user's name
- (2) if "hlq" is keyword R or K, 
  - set the flag
- (3) if "tmphlq" is defined, 
  -  default behavior, same as from the original `if`.

So what if someone wants their restored data sets to have hlq `ABCD`, and thus defines `hlq: ABCD`, but say the temp data sets should have hlq `XYZ`, so `tmp_hlq: XYZ`. Running through the conditional above, `hlq` is not none, so (1) gets skipped, `hlq` is not 'R' or 'K' so (2) gets skipped. `tmp_hlq` is defined, so now `zoau_args["high_level_qualifier"]` gets set to the user name?

Reply:
Good catch, I already make some changes for those cases return the default_hlq to default value for the hlq and also the if check first exist and then see if is a r. And now I'm using and r because or validation required the hlq should be in upper case, by using only one case of lower case could work as keyword.